src/tools/rbd/action/Snap.cc

```diff
@@ -399,11 +411,23 @@ int execute_unprotect(const po::variables_map &vm) {
   if (r < 0) {
     return r;
   }
-
-  r = do_unprotect_snap(image, snap_name.c_str());
+  
+  bool is_protected = false;
+  r = image.snap_is_protected(snap_name.c_str(), &is_protected);
   if (r < 0) {
-    std::cerr << "rbd: unprotecting snap failed: " << cpp_strerror(r)
-              << std::endl;
+    return r;
+  } 
```

Comment:
I would move the unprotect log message logic here: } else if (!protected) { ... };